src/Symfony/Component/DependencyInjection/ReverseContainer.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException;
+
+/**
+ * Turns public and "container.reversible" services back to their ids.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+final class ReverseContainer
+{
+    private $serviceContainer;
+    private $reversibleLocator;
+    private $getServiceId;
+
+    public function __construct(Container $serviceContainer, ContainerInterface $reversibleLocator, string $tagName = 'container.reversible')
+    {
+        $this->serviceContainer = $serviceContainer;
+        $this->reversibleLocator = $reversibleLocator;
+        $this->getServiceId = \Closure::bind(function ($service) {
+            return array_search($service, $this->services, true) ?: array_search($service, $this->privates, true);
+        }, $serviceContainer, Container::class);
+    }
+
+    /**
+     * @param object $service
+     *
+     * @throws ServiceNotFoundException When the service is not reversible and $throw is true
+     */
+    public function getId($service, bool $throw = true): ?string
```

Comment:
I removed the argument.

Reply:
Dropping the argument looks good to me (you may want to update the PR description now). But doesn't it make more sense to always throw instead of returning `null`?

Reply:
Both use cases exist, that's why I added the argument first: by default throwing looks good to me. But throwing is slow, which means the reverse container wouldn't be usable in a situation where you want to quickly check if an object is a reversible service - e.g. in a Serializer normalizer.

Reply:
description updated